src/compas_fab/robots/robot.py

```diff
@@ -110,6 +112,49 @@ def __from_data__(cls, data):
         robot.attributes = attributes
         return robot
 
+    @classmethod
+    def from_urdf(cls, urdf_filename, srdf_filename=None, local_package_mesh_folder=None, client=None):
+        # type: (str, Optional[str], Optional[str], Optional[ClientInterface]) -> Robot
+        """Create a robot from URDF.
+        Optionally, SRDF can be provided to load semantics and a local package mesh folder to load mesh geometry.
+
+        Parameters
+        ----------
+        urdf_filename : :obj:`str`
+            Path to the URDF file.
+        srdf_filename : :obj:`str`, optional
+            Path to the SRDF file to load semantics. Default is `None`.
+        local_package_mesh_folder : :obj:`str`, optional
+            Path to the local package mesh folder.
+            If the path is provided, the geometry of the robot is loaded from this folder.
+            Default is `None`, which means that the geometry is not loaded.
+        client : :class:`compas_fab.backends.interfaces.ClientInterface`, optional
+            Backend client provided for the . Default is `None`.
+
+        Returns
+        -------
+        :class:`compas_fab.robots.Robot`
+            Newly created instance of the robot.
+
+        """
+        # NOTE: This import is here to avoid circular imports
+        from compas_fab.robots import RobotSemantics
+
+        model = RobotModel.from_urdf_file(urdf_filename)
+
+        if srdf_filename:
+            semantics = RobotSemantics.from_srdf_file(srdf_filename, model)
+        else:
+            semantics = None
```

Comment:
Maybe this is a bit neater?

```suggestion
        semantics = None

        if srdf_filename:
            semantics = RobotSemantics.from_srdf_file(srdf_filename, model)
```